### PR TITLE
chore(ssa refactor): Add code to handle `truncate` instruction

### PR DESCRIPTION
**Pull request by kevaundray**
# Description

## Problem\*

Resolves 

## Summary\*

This PR sets out to

### Example

Before:

```

```

After:

```

```

## Documentation

- [ ] This PR requires documentation updates when merged.

  

  - [ ] I will submit a noir-lang/docs PR.

  

  - [ ] I will request for and support Dev Rel's help in documenting this PR.

  
  

## Additional Context

# PR Checklist\*

- [ ] I have tested the changes locally.
- [ ] I have formatted the changes with [Prettier](https://prettier.io/) and/or `cargo fmt` on default settings.

### CODE REVIEW

**Comment by kevaundray** [comment]
Currently the SSA IR does not spit out truncate instructions, so acir gen will call the underlying `truncate_var` method when truncation is needed. For the moment, it will call it excessively instead, we _could_ add the analysis in AcirContext, but this is not needed right now

File: crates/noirc_evaluator/src/ssa_refactor/acir_gen/acir_ir/acir_variable.rs [hunk 1]
```diff
@@ -431,6 +431,20 @@ impl AcirContext {
         Ok(variable)
     }
 
+    /// Returns an `AcirVar` which will be constrained to be lhs mod 2^{rhs}
```
Comment (ludamad) [inline crates/noirc_evaluator/src/ssa_refactor/acir_gen/acir_ir/acir_variable.rs:434]: nit: NAND cspell

File: crates/noirc_evaluator/src/ssa_refactor/acir_gen/acir_ir/generated_acir.rs [hunk 2]
```diff
@@ -86,6 +87,59 @@ impl GeneratedAcir {
...
+        let c_arith = Expression::from(c_witness);
+
+        let res = &b_arith + &(two_pow_rhs_bits * &c_arith);
+        let my_constraint = &res - lhs;
+
+        self.push_opcode(AcirOpcode::Arithmetic(my_constraint));
+
+        Ok(Expression::from(b_witness))
+    }
+
```
Comment (ludamad) [inline crates/noirc_evaluator/src/ssa_refactor/acir_gen/acir_ir/generated_acir.rs:142]: nit: booleanity cspell (maybe just => boolean)

**Comment by ludamad** [comment]
LGTM

**Review by ludamad — APPROVED** [review]

File: crates/noirc_evaluator/src/ssa_refactor/acir_gen/acir_ir/acir_variable.rs (hunk 1 above)
Comment (kevaundray) [reply crates/noirc_evaluator/src/ssa_refactor/acir_gen/acir_ir/acir_variable.rs:434]: Created #1533 to resolve this

File: crates/noirc_evaluator/src/ssa_refactor/acir_gen/acir_ir/generated_acir.rs (hunk 2 above)
Comment (kevaundray) [reply crates/noirc_evaluator/src/ssa_refactor/acir_gen/acir_ir/generated_acir.rs:142]: Created #1533 to resolve this

**Comment by jfecher** [comment]
> For the moment, it will call it excessively instead,

It doesn't look like it is being called at all currently

**Pull request merged**